File upload

semohr opened the pull request:
Adds file upload capabilities to our inbox view. 

ToDos:
- [x] Frontend component drag/drop files into inbox
- [x] Action to trigger upload without drag and drop
- [x] Backend upload endpoint 
- [x] ~~Currently, the PR breaks the dev setup - errors on dynamic reloads~~ (seems to only occur when working on the dialog or dropzone, PS I think thats tolerable)
- [ ] docs (not the limit about folders, ssl required)
- [x] error handling?
- [x] test for backend route

semohr committed:
Added a minimal file upload form the the frontend. Allows to select files
via drag and drop into the browser window or by manual selection.

pSpitzner commented:
#102

Really loving this! 
While we are at it, lets add a simple info when dragging in a folder (since thats likely a common use-case):
- number music files inside
- number non-music files
- number folders

semohr commented:
Have never worked with directories but seems [doable](https://web.dev/patterns/files/drag-and-drop-directories).

Side note: All file access apis are only available in the secure context (https).

pSpitzner committed:
wip: implemented file drag example (frontend) and added backend route.
tests now run through, but it still seems that our custom exceptions do not
 propagate through the xhr (invalid targetdir does not raise in frontend nor console log)

pSpitzner committed:
added new dropzone animation. not quite there yet, but i like the different inbox styling.

pSpitzner committed:
drag prevention on outside of inbox dorps

pSpitzner committed:
fixed drop not releasing drag overlay

pSpitzner committed:
added progress overlay

pSpitzner committed:
prevent uploading anything thats not a single archive file (only via frontend)

pSpitzner committed:
reworked upload, now supports multiple files, progress, and error handling.
TODO: mobile, use dialog, folder creation for multiple files, snackbar.

Co-Authored-By: semohr <sebastian@mohrenclan.de>

semohr committed:
Quite some cleanup:
- Added generic useDrag hook
- Refactored upload into multiple files

Co-authored-by: pSpitzner <github@makeitso.one>

semohr committed:
Finalized and styled upload progress for files.

Other dialog parts still need some work but basic functions
are working.

Co-authored-by: pSpitzner <github@makeitso.one>

pSpitzner committed:
a few changes:
- dialog styling
- enabled dialog dropzone
- excluding .git from repo permission fix

a few bugs around uploading:
- we can add the same file multiple times
- we can add files after the upload finished :P

pSpitzner committed:
a bit of progress on what to do after uploading. (resetting progress)

pSpitzner committed:
added cancel button and automatic dialog closing

semohr committed:
Added upload temp_dir configuration for imbox upload.

semohr committed:
Removed unnecessary title state, should be derived.

semohr committed:
Moved filelist and dir out of api layer and refactored state slightly.
Structure is now flat instead of nested. Makes things easier.

semohr committed:
Filename and dirs are not parsed correctly because of url encoding.

semohr committed:
Readded close form after timeout.

semohr committed:
Enhanced file upload tests with some extra cases.

semohr committed:
Fixed ci pipeline

pSpitzner committed:
run prettier

pSpitzner committed:
Added Upload Action Button

pSpitzner committed:
Added some tests for the validation route

pSpitzner commented:
> Adds file upload capabilities to our inbox view.
> 
> ToDos:
> 
> * [x]  Frontend component drag/drop files into inbox
> * [x]  Action to trigger upload without drag and drop
> * [x]  Backend upload endpoint
> * [ ]  Currently, the PR breaks the dev setup - errors on dynamic reloads
> * [ ]  docs (not the limit about folders, ssl required)
> * [x]  error handling?
> * [x]  test for backend route

We already had most of the needed tests, also, error handling seems to work:
![screenshot_2025-12-26_214617@2x](https://github.com/user-attachments/assets/efd44363-5a89-4ca7-91ad-55aa924021fa)

pSpitzner committed:
removed legacy file upload frontend component

pSpitzner committed:
re-added config option for upload dir, frontend: added upload section to actions-info
